components/patient-data/impl/src/main/java/org/phenotips/data/internal/PhenoTipsPatientRepository.java

```diff
@@ -204,4 +206,31 @@ private long getLastUsedId() throws QueryException
         crtMaxID = Math.max(crtMaxID, 0);
         return crtMaxID;
     }
+
+    @Override
+    public Collection<Patient> getAllPatients()
+    {
+        List<String> patientIds = null;
+        try {
+            Query q = this.qm.createQuery(
+                "select doc.name "
+                + "from Document doc, "
+                + "doc.object(PhenoTips.PatientClass) as patient "
+                + "where patient.identifier is not null ", Query.XWQL);
```

Comment:
In PhenomeCentral we have a convention, if `patient.identifier < 0` that's not a real patient, but an OMIM template. Although that convention is temporary, at the moment you should also put that condition.

    where patient.identifier is not null

becomes

    where patient.identifier >= 0